readthedocs/search/documents.py

```diff
@@ -130,6 +109,23 @@ class Meta:
         fields = ('commit', 'build')
         ignore_signals = True
 
+    def prepare_domains(self, html_file):
+        """Prepares and returns the values for domains field."""
+        domains_qs = html_file.sphinx_domains.all()
```

Comment:
This should probably use `iterator()` for memory usage issues. I don't expect too many results here, but good to be safe.